src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> oldLast = last;
+        Node<T> newLast = new Node<>(oldLast, value, null);
+        last = newLast;
+        if (oldLast == null) {
+            first = newLast;
+        } else {
+            oldLast.next = newLast;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexPossibility(index);
+        if (index == size) {
+            add(value);
+        } else {
+            linkBefore(value, getNodeByIndex(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            throw new RuntimeException("The list is empty!");
+        }
+        for (var i : list) {
+            this.add(i);
```

Comment:
```suggestion
        for (var i : list) {
            add(i);
```
i is informative name?